Does #712  (Use only pragma once and no more include guards) break many NEURON models?

**Describe the issue**
Latest NEURON master does not compile with latest CoreNEURON  master when NEURON CMake configured ```-DNRN_ENABLE_CORENEURON=ON```.  ```nrn/src/nrnoc/netstim.mod``` and ```coreneuron/mechanism/mech/modfile/netstim.mod``` contain the lines
```
VERBATIM
#include "nrnran123.h"
```
and mod2c on translation adds the line to nrn/build/bin/x86_64/corenrn/mod2c/netstim.cpp
```
#include "coreneuron/utils/randoms/nrnran123.h"
```
So during build time I see errors like:
```
x86_64/corenrn/mod2c/netstim.cpp:624:48: error: call of overloaded ‘nrnran123_newstream3(uint32_t&, uint32_t&, uint32_t&)’ is ambiguous
  *pv = nrnran123_newstream3(di[0], di[1], di[2]);
                                                ^
In file included from x86_64/corenrn/mod2c/netstim.cpp:423:0:
/home/hines/neuron/before-after/build/include/nrnran123.h:46:25: note: candidate: nrnran123_State* nrnran123_newstream3(uint32_t, uint32_t, uint32_t)
 extern nrnran123_State* nrnran123_newstream3(uint32_t id1, uint32_t id2 , uint32_t id3);
                         ^~~~~~~~~~~~~~~~~~~~
In file included from x86_64/corenrn/mod2c/netstim.cpp:12:0:
/home/hines/neuron/before-after/build/include/coreneuron/utils/randoms/nrnran123.h:85:18: note: candidate: coreneuron::nrnran123_State* coreneuron::nrnran123_newstream3(uint32_t, uint32_t, uint32_t, bool)
 nrnran123_State* nrnran123_newstream3(uint32_t id1,
                  ^~~~~~~~~~~~~~~~~~~~
```
The problem goes away if the old guard is added back to coreneuron/utils/randoms/nrnran123.h or that file is reverted from 
```
$ git show 0f071d33 nrnran123.h
commit 0f071d33752810ebfb273ea0b8db436c6d5d4c58
Author: Nicolas Cornu <nicolas.cornu@epfl.ch>
Date:   Thu Jan 6 11:20:22 2022 +0100

    Use only pragma once and no more include guards (#712)
    
    * There were 30 files with pragma once and 42 with include guards
       Now use pragma once everywhere
    * Include only source headers for coreneuron and nrnivmodl-core
      uses headers from build directory only

diff --git a/coreneuron/utils/randoms/nrnran123.h b/coreneuron/utils/randoms/nrnran123.h
index c9759216..27335f77 100644
--- a/coreneuron/utils/randoms/nrnran123.h
+++ b/coreneuron/utils/randoms/nrnran123.h
@@ -7,8 +7,7 @@
 */
 // Beware changing this to #pragma once, we rely on this file shadowing the
 // equivalent file from NEURON.
-#ifndef nrnran123_h
-#define nrnran123_h
+#pragma once
 
 /* interface to Random123 */
 /* http://www.thesalmons.org/john/random123/papers/random123sc11.pdf */
@@ -128,5 +127,3 @@ CORENRN_HOST_DEVICE_ACC nrnran123_array4x32 nrnran123_iran(uint32_t seq,
 CORENRN_HOST_DEVICE_ACC double nrnran123_uint2dbl(uint32_t);
 nrn_pragma_omp(end declare target)
 }  // namespace coreneuron
-
-#endif
```

**To Reproduce**
```
git clone git@github.com:neuronsimulator/nrn nrn
cd nrn
git checkout master
mkdir build
cd build
cmake .. -G Ninja -DCMAKE_INSTALL_PREFIX=install -DPYTHON_EXECUTABLE=`which python3` -DNRN_ENABLE_RX3D=OFF -DCMAKE_BUILD_TYPE=Debug -DNRN_ENABLE_TESTS=ON -DNRN_ENABLE_CORENEURON=ON
(cd ../external/coreneuron; git checkout master; git pull)
ninja
```

--- a/.github/workflows/test-as-submodule.yml
+++ b/.github/workflows/test-as-submodule.yml
@@ -1,0 +1,116 @@
+name: NEURON submodule
+
+concurrency:
+  group: ${{ github.workflow }}#${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - master
+      - release/**
+  pull_request:
+    branches:
+      - master
+      - release/**
+
+jobs:
+  ci:
+    name: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            cores: 2
+          - os: macOS-11
+            cores: 3
+      fail-fast: false
+    env:
+      CMAKE_BUILD_PARALLEL_LEVEL: ${{matrix.cores}}
+    steps:
+      - name: Install homebrew packages
+        if: startsWith(matrix.os, 'macOS')
+        run: |
+          brew install bison coreutils flex ninja openmpi
+          python3 -m pip install --upgrade numpy pytest pytest-cov
+          echo /usr/local/opt/flex/bin:/usr/local/opt/bison/bin >> $GITHUB_PATH
+          echo "CC=clang" >> $GITHUB_ENV
+          echo "CXX=clang++" >> $GITHUB_ENV
+
+      - name: Install apt packages
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt-get install bison cython3 flex libfl-dev libopenmpi-dev \
+            ninja-build openmpi-bin python3-dev python3-numpy python3-pytest \
+            python3-pytest-cov
+          echo "CC=gcc" >> $GITHUB_ENV
+          echo "CXX=g++" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v2
+        name: Checkout NEURON
+        with:
+          # TODO: allow configuring a non-default branch of NEURON, ideally
+          #       using the same CI_BRANCHES syntax in a PR body that is used
+          #       by the GitLab CI plans.
+          path: nrn
+          repository: neuronsimulator/nrn
+
+      - name: Update CoreNEURON submodule
+        run: |
+          cd ${GITHUB_WORKSPACE}/nrn
+          coreneuron_sha=${{github.event.pull_request.head.sha}}
+          if [[ -z ${coreneuron_sha} ]]; then
+          # presumably we're running on a push event
+          coreneuron_sha=${{github.sha}}
+          fi
+          echo "Using CoreNEURON SHA ${coreneuron_sha}"
+          # https://stackoverflow.com/a/33575837
+          git update-index --cacheinfo 160000,${coreneuron_sha},external/coreneuron
+          git submodule update --recursive
+          echo "NEURON status"
+          git status
+          git log -n 1
+          cd external/coreneuron
+          echo "CoreNEURON status"
+          git status
+          git log -n 1
+
+      - name: Configure NEURON
+        run: |
+          cd ${GITHUB_WORKSPACE}/nrn
+          mkdir build install
+          cd build
+          # NEURON CMake assumes this is defined.
+          export SHELL=$(command -v bash)
+          cmake .. -G Ninja \
+            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+            -DCMAKE_INSTALL_PREFIX=../install \
+            -DPYTHON_EXECUTABLE=$(command -v python3) \
+            -DNRN_ENABLE_CORENEURON=ON \
+            -DNRN_ENABLE_INTERVIEWS=OFF \
+            -DNRN_ENABLE_RX3D=OFF \
+            -DNRN_ENABLE_TESTS=ON
+
+      - name: Build NEURON
+        run: |
+          cd ${GITHUB_WORKSPACE}/nrn/build
+          cmake --build . --parallel
+
+      - name: Test NEURON
+        run: |
+          cd ${GITHUB_WORKSPACE}/nrn/build
+          ctest --output-on-failure
+
+      - name: Install NEURON
+        run: |
+          cd ${GITHUB_WORKSPACE}/nrn/build
+          cmake --build . --target install
+
+      # This step will set up an SSH connection on tmate.io for live debugging.
+      # To enable it, you have to:
+      #   * add 'live-debug-ci' to your PR title
+      #   * push something to your PR branch (note that just re-running the pipeline disregards the title update)
+      - name: live debug session on failure (manual steps required, check `.github/workflows/test-as-submodule.yml`)
+        if: failure() && contains(github.event.pull_request.title, 'live-debug-ci')
+        uses: mxschmitt/action-tmate@v3

--- a/extra/nrnivmodl_core_makefile.in
+++ b/extra/nrnivmodl_core_makefile.in
@@ -49,7 +49,9 @@ CORENRNLIB_FLAGS += $(if @caliper_LIB_DIR@, -W$(subst ;, -W,l,-rpath,@caliper_LI
 CORENRNLIB_FLAGS += $(if @caliper_LIB_DIR@,-L@caliper_LIB_DIR@,)
 
 # Includes paths gathered by CMake
-INCLUDES = $(INCFLAGS) -I$(CORENRN_INC_DIR) -I$(CORENRN_INC_DIR)/coreneuron/utils/randoms
+# coreneuron/utils/randoms goes first because it needs to override the NEURON
+# directory in INCFLAGS
+INCLUDES = -I$(CORENRN_INC_DIR)/coreneuron/utils/randoms $(INCFLAGS) -I$(CORENRN_INC_DIR)
 INCLUDES += $(if @MPI_CXX_INCLUDE_PATH@, -I$(subst ;, -I,@MPI_CXX_INCLUDE_PATH@),)
 INCLUDES += $(if @reportinglib_INCLUDE_DIR@, -I$(subst ;, -I,@reportinglib_INCLUDE_DIR@),)
 

